app/controllers/ems_common.rb

```diff
@@ -631,36 +633,76 @@ def deleteemss
     end
   end
 
-  def call_ems_refresh(emss)
-    process_emss(emss, "refresh_ems") unless emss.empty?
+  def call_ems_task(emss, task, action_name)
+    return unless emss.any?
+    if task == "refresh_ems"
+      model.refresh_ems(emss, true)
+      add_success_flash_and_audit(emss, task)
+    elsif task == "capture_ems"
+      model.where(:id => emss).each(&:queue_metrics_capture)
+      add_success_flash_and_audit(emss, task)
+    end
+
     return if @flash_array.present?
 
-    add_flash(n_("Refresh initiated for %{count} %{model} from the %{product} Database",
-                 "Refresh initiated for %{count} %{models} from the %{product} Database", emss.length) %
-      {:count   => emss.length,
-       :product => Vmdb::Appliance.PRODUCT_NAME,
-       :model   => ui_lookup(:table => table_name),
-       :models  => ui_lookup(:tables => table_name)})
+    add_flash(n_("%<action_name>s initiated for %<count>d %<model>s from the %<product>s Database",
+                 "%<action_name>s initiated for %<count>d %<models>s from the %<product>s Database", emss.length) %
```

Comment:
isn't this flash already displayed by `add_success_flash_and_audit()` calls above?